core/domain/blog_services.py

```diff
@@ -0,0 +1,613 @@
+        blog_post_summary: BlogPostSummary. The summary object for the given
+            blog post summary.
+    """
+    model = blog_models.BlogPostSummaryModel.get(
+        blog_post_summary.id, strict=False)
+    if model:
+        model.author_id = blog_post_summary.author_id
+        model.title = blog_post_summary.title
+        model.summary = blog_post_summary.summary
+        model.tags = blog_post_summary.tags
+        model.published_on = blog_post_summary.published_on
+        model.thumbnail_filename = blog_post_summary.thumbnail_filename
+        model.url_fragment = blog_post_summary.url_fragment
+    else:
+        model = blog_models.BlogPostSummaryModel(
+            id=blog_post_summary.id,
+            author_id=blog_post_summary.author_id,
+            title=blog_post_summary.title,
+            summary=blog_post_summary.summary,
+            tags=blog_post_summary.tags,
+            published_on=blog_post_summary.published_on,
+            thumbnail_filename=blog_post_summary.thumbnail_filename,
+            url_fragment=blog_post_summary.url_fragment,
+        )
+    model.update_timestamps()
+    model.put()
+
+
+def save_blog_post_rights(blog_post_rights):
+    """Saves a BlogPostRights domain object to the datastore.
+
+    Args:
+        blog_post_rights: BlogPostRights. The rights object for the given
+            blog post.
+    """
+    model = blog_models.BlogPostRightsModel.get(
+        blog_post_rights.id, strict=True)
+
+    model.editor_ids = blog_post_rights.editor_ids
+    model.blog_post_is_published = blog_post_rights.blog_post_is_published
+    model.update_timestamps()
+    model.put()
+
+
+def check_can_edit_blog_post(user, blog_post_rights):
+    """Checks whether the user can edit the given blog post.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        blog_post_rights: BlogPostRights or None. Rights object for the given
+            blog post.
+
+    Returns:
+        bool. Whether the given user can edit the given blog post.
+    """
+    if blog_post_rights is None:
+        return False
+    if role_services.ACTION_EDIT_ANY_BLOG_POST in user.actions:
+        return True
+    if blog_post_rights.is_editor(user.user_id):
+        return True
+
+    return False
+
+
+def deassign_user_from_all_blog_posts(user_id):
+    """Removes the user from the list of editor_ids for all blog posts.
+
+    Args:
+        user_id: str. ID to be removed from editor_ids.
+    """
+    blog_models.BlogPostRightsModel.deassign_user_from_all_blog_posts(
+        user_id)
+
+
+def generate_url_fragment(title):
+    """Generates the url fragment for a blog post from the title of the blog
+    post.
+
+    Args:
+        title: str. The title of the blog post.
+
+    Returns:
+        str. The url fragment of the blog post.
+    """
+    lower_title = title.lower()
+    hyphenated_title = lower_title.replace(' ', '-')
+    return hyphenated_title
```

Comment:
Are we expecting to have a "random" value at the end of the URL fragment? (not sure!)

Reply:
Done. Before I was thinking of adding the blog post ID in the front end to the blog post url, but we can add it here as well.